activesupport/lib/active_support/notifications.rb

```diff
@@ -162,6 +163,28 @@ def instrument(name, payload = {})
         end
       end
 
+      # Requires an array of paths, which correspond to the files and
+      # directories to publish file changes from. All files within the
+      # specified directories will be watched (including within recursive
+      # subdirectories).
+      #
+      # Only a single array of paths can be watched at a time. Calling this
+      # method a second time will stop all notifications about the file
+      # changes from the set of paths from the first call to the method.
+      def publish_file_changes(name, paths, opts={})
```

Comment:
I don't think these methods belongs to this class. This class is for generic instrumentation, we should not include specific tasks like watch the file system. Maybe we should create a subclass of instrumentation.

cc @tenderlove